string/string.go

```diff
@@ -0,0 +1,155 @@
+package string
+
+import (
+	"strings"
+)
+
+// ToLower converts a string to lowercase.
+func ToLower(s string) string {
+	return strings.ToLower(s)
+}
+
+// ToUpper converts a string to uppercase.
+func ToUpper(s string) string {
+	return strings.ToUpper(s)
+}
+
+// Concat concatenates two strings.
+func Concat(s1, s2 string) string {
+	return s1 + s2
+}
+
+// Reverse reverses a string.
+func Reverse(s string) string {
+	// Convert the string to a rune slice to handle Unicode characters correctly
+	runes := []rune(s)
+	length := len(runes)
+	reversed := make([]rune, length)
+
+	for i, j := length-1, 0; i >= 0; i, j = i-1, j+1 {
+		reversed[j] = runes[i]
```

Comment:
may be avoid creating a reversed rune array for a new length and instead use 

 for i, j := 0, len(runes)-1; i < j; i, j = i+1, j-1 { 
         runes[i], runes[j] = runes[j], runes[i] 
    } 

 return string(runes)